crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -6626,4 +6656,92 @@ mod tests {
 
         Ok(())
     }
+
+    /// This test verifies that queries
+    #[test]
+    fn dependency_own_instance_member() -> anyhow::Result<()> {
```

Comment:
the amount of setup required here makes me think that we might want to prioritise adding support for incremental tests to mdtest, similar to mypy's test framework (take a look at https://github.com/python/mypy/blob/master/test-data/unit/check-incremental.test)

Reply:
Yeah, although copy pasting is fine, considering how few we have but it could be something to consider for a more scalable approach on how to avoid cross-file queries.

Reply:
yeah, it would only be worth it if we consider that it would be useful to have a lot more tests like this, and we want to make it easier to write those tests.

Reply:
There was already a design for this in the original mdtest design (and it's in the README as a planned improvement), we just haven't implemented yet. (And the design probably needs to change a bit to avoid using non-rendering tag info strings on code blocks.)